server/src/Utils/Order.hs

```diff
@@ -0,0 +1,38 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Utils.Order where
+
+import Import.NoFoundation
+import Model.Types
+
+-- {-| Determine if a given user is allowed to access the PDF file of an item.
+-- -}
+hasAccessToPdfFileDownload ::
+       ( BaseBackend (YesodPersistBackend site) ~ SqlBackend
+       , YesodPersist site
+       , PersistQueryRead (YesodPersistBackend site)
+       )
+    => UserId
+    -> Text
+    -> HandlerT site IO AuthResult
+hasAccessToPdfFileDownload userId filename = do
+    let unauthorized = Unauthorized "You didn't buy this item"
```

Comment:
The problem is that `selectFirst :: (PersistEntity val, PersistEntityBackend val ~ b) => [Filter val] -> [SelectOpt val] -> b m (Maybe (Entity val))`. So we have a `Maybe (Entity val)` which I need to take care of.